Added kms keys and tags to module.

kalpatel01 opened the pull request:
Added 2 features to ec2_vol module:
- kms key support
- ability to add tags (Module exist to add and remove tags, adding here makes dynamic tagging easier)

Fixed bug where AnsibleAWSError doesn't exist.

underyx committed:
Add prefixing and suffixing fuctionality to assemble

zarnovican committed:
Py2.4: SystemExit in async_wrapper is not an error - compatibility fix

Prior to Python 2.5, SystemExit was a subclass of Exception.
In Py2.4, this is causing extra error output on valid sys.exit(0).

(Toshio) Call sys.exit from inside of the SystemExit exception handler so py2.4 and py2.5+ behaviour matches

Commit:
Added kms keys and tags to module.

jmunhoz committed:
Add Ceph RGW S3 compatibility

Ceph Object Gateway (Ceph RGW) is an object storage interface built on top of
librados to provide applications with a RESTful gateway to Ceph Storage
Clusters:

http://docs.ceph.com/docs/master/radosgw/

This patch adds the required bits to use the RGW S3 RESTful API properly.

Signed-off-by: Javier M. Mellid <jmunhoz@igalia.com>

robynbergeron commented:
Thanks @kalpatel01 for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Commit:
Fixed version number and if kms key is defined will automatically set the encrypted flag.

Commit:
Fixed description

mpappas86 committed:
Prevent ec2group from deleting sgs during check_mode runs

dougluce committed:
Fix S3 unavailable region error

This is to address this error:

  fatal: [site]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to connect to S3: Region  does not seem to be available for awsmodule boto.s3. If the region definitely exists, you may need to upgrade boto or extend with endpoints_path"}

Commit 0dd58e9 changed the logic so an exception is thrown (by
`connect_to_aws`) before the `s3 is None` check is performed. This
changes the `None` check to a catch so the old logic can compensate.

Commit:
fix #3528 : dns type

Commit:
Add sc1 and st1 volume types to ec2_vol.py

AWS just added two new volume types.

https://aws.amazon.com/blogs/aws/amazon-ebs-update-new-cold-storage-and-throughput-options/

Commit:
Fixed description with 2 new volume types.

dagwieers committed:
Add unpack results to error output, for debugging purposes on failure

This may help understand what is going on with #3631

privateip committed:
Merge pull request #3693 from keinohguchi/json_non_string_key_fix

ops_template.py: Change integer key into stirng

mscherer committed:
Convert the network subfolder to py3/py2.4 syntax (#3690)

Commit:
Where to get the full docker picture.

Commit:
Make the new bits a NOTE

Commit:
Uppercase

Commit:
Merge pull request #3697 from chouseknecht/docker-docs

Make sure users know where to find the full docker picture.

gillesgagniard committed:
Initial support for specifying in which subnetwork a gce instance should be created. This is required for non-legacy networks.

gillesgagniard committed:
fix documentation to reflect subnetwork option is new in ansible 2.2

Commit:
Remove the U().

Commit:
Merge pull request #3704 from chouseknecht/devel

Remove the U().

Commit:
Fixed the URL in description. It's now valid Yaml.

Commit:
Merge pull request #3705 from chouseknecht/devel

Fixed the URL in description. It's now valid Yaml.

chlam4 committed:
Added support for accepting extra config options in vm reconfiguration. (#3327)

abadger committed:
Py3 exclude list (#3698)

* Fix a few modules to pass syntax checks under python3

* Move from a whitelist of modules to check for python3 compat to a blacklist

jsternberg committed:
Paginate the results from `boto.iam`

The default pagination is every 100 items with a maximum of 1000 from
Amazon. This properly uses the marker returned by Amazon to concatenate
the various pages from the results.

This fixes #2440.

jctanner committed:
Do not ignore the primary group if modifying the list of secondary groups. (#3585)

Fixes #1118

sivel committed:
2 Module fixes (#3611)

* Fix syntax error in azure_rm_virtualmachine.py

* Allow rhn_register to fail gracefully when rhn-client-tools is not installed

lloydhazlett committed:
Add path type to argument_spec where applicable (#3040)

gillesgagniard committed:
better error message if libcloud is not recent enough to use gce subnetworks

kwerey committed:
Add option to make RDS param groups for the MySQL 5.7 engine family. (#3695)

nbrownus committed:
Do not expect the elb to have policies or instances. Fixes #3593 (#3595)

ryansb committed:
Allow users to pick AWS Aurora as an RDS engine (#3414)

Per the [RDS docs](http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html),
add `aurora` as a valid DB engine.

Commit:
Unindenting the delete to user (#3147)

Jmainguy commented on cloud/amazon/ec2_vol.py:
```diff
@@ -100,6 +100,18 @@
+        e.g.: arn:aws:kms:us-east-1:012345678910:key/abcd1234-a123-456a-a12b-a123b4cd56ef
+    required: false
+    default: None
+    version_added: 2.39
```
This should be the ansible version it got merged in under, not the boto version. Change this to 2.2 for example as thats likely the version this would get merged into.

mscherer committed:
Convert packaging/* to py3/py2.4 syntax (#3702)

Jmainguy commented on cloud/amazon/ec2_vol.py:
```diff
@@ -100,6 +100,18 @@
+    description:
+      - add additional tags to a volume.
+    required: false
+    default: null
```
Should add version_added: here as well.

mscherer committed:
Improve error reporting when git binary fail (#3266)

Showing stderr or stdout is a great help to understand
when something go south.

isopy commented:
I really need this functionality for a project I'm working so I tested this PR using the following task (with boto 2.40):

```
ec2_vol:
  instance: "{{ item.id }}"
  device_name: /dev/xvdf
  volume_type: gp2
  volume_size: 30
   delete_on_termination: true
  encrypted: true
  kms_key_id: "arn:aws:kms:us-west-2:############:key/##-##-##-##-#####"
with_items: ec2.instances
```

I'm getting the following error:

```
failed: [hostname] (item={u'kernel': None, u'root_device_type': u'ebs', u'private_dns_name': u'ip-##-##-##-##-us-west-2.compute.internal', u'public_ip': None, u'private_ip': u'##.##.##.##', u'id': u'i-b563b373', u'ebs_optimized': False, u'state': u'running', u'virtualization_type': u'hvm', u'architecture': u'x86_64', u'ramdisk': None, u'block_device_mapping': {u'/dev/sda1': {u'status': u'attached', u'delete_on_termination': True, u'volume_id': u'vol-c5c6ac73'}}, u'key_name': u'shared', u'image_id': u'ami-######', u'tenancy': u'default', u'groups': {u'sg-#######': u'ZZZInternalSg2', u'sg-######': u'ZZZ-DefaultSg-1PD3EEFE9PI47', u'sg-######': u'ZZZInternalSg1', u'sg-######': u'ZZZInternalSg3'}, u'public_dns_name': u'', u'state_code': 16, u'tags': {u'Name': u'ZZZMeta02.dw', u'ServicePhase': u'Prod', u'ServiceComponent': u'ZZZMeta', u'clusterid': u'', u'ServiceName': u'ZZZ', u'host-type': u'', u'ServiceOwner': u'ZZ'}, u'placement': u'us-west-2a', u'ami_launch_index': u'0', u'dns_name': u'', u'region': u'us-west-2', u'launch_time': u'2016-05-20T15:43:23.000Z', u'instance_type': u'r3.large', u'root_device_name': u'/dev/sda1', u'hypervisor': u'xen'}) => {"failed": true, "item": {"ami_launch_index": "0", "architecture": "x86_64", "block_device_mapping": {"/dev/sda1": {"delete_on_termination": true, "status": "attached", "volume_id": "vol-c5c6ac73"}}, "dns_name": "", "ebs_optimized": false, "groups": {"sg-1885637e": "ZZZInternalSg3", "sg-1985637f": "ZZZInternalSg2", "sg-1d85637b": "ZZZInternalSg1", "sg-b79af0d2": "ZZZ-DefaultSg-#####"}, "hypervisor": "xen", "id": "i-b563b373", "image_id": "ami-#####", "instance_type": "r3.large", "kernel": null, "key_name": "shared", "launch_time": "2016-05-20T15:43:23.000Z", "placement": "us-west-2a", "private_dns_name": "ip-#####.us-west-2.compute.internal", "private_ip": "#####", "public_dns_name": "", "public_ip": null, "ramdisk": null, "region": "us-west-2", "root_device_name": "/dev/sda1", "root_device_type": "ebs", "state": "running", "state_code": 16, "tags": {"Name": "ZZZMeta02.dw", "ServiceComponent": "ZZZMeta", "ServiceName": "ZZZ", "ServiceOwner": "ZZZ", "ServicePhase": "Prod", "clusterid": "", "host-type": ""}, "tenancy": "default", "virtualization_type": "hvm"}, "msg": "InvalidVolume.NotFound: The volume 'vol-20c6ac96' does not exist."}
```

The task works fine when I comment out the kms_key_id.

privateip committed:
Merge pull request #3623 from ocadotechnology/wip-3621

junos_template: Simplify arguments. Fixes #3621

mscherer committed:
Fix wait_for to compile on python3 (#3722)

mscherer committed:
Port vsphere_guest to py3/py2.4 syntax (#3721)

kenwmitchell committed:
Proposed fix to not require src and fstype arguments for mount. Issue #1259. (#1357)

jtyr committed:
Removing double definition in postgresql_user module (#3714)

This commit is removing double definition of variables in the
grant_privileges function of the postgresql_user module.

marcosdiez committed:
The documentation of lineinfile now makes reference to blockinfile (#3639)

marcosdiez committed:
Added one extra example in the authorized_key module. (#3637)

ryansb committed:
Merge pull request #3320 from hyperized/patch-1

Added restart functionality to ec2.py

Commit:
Make add ip_floating_ip idempotent

Closes issue #1858

james-prior committed:
Make explicit that virtualenv is created if needed for pip module. (#3731)

packaging/language/pip.py:
    virtualenv option:
        Mention that virtualenv is created if it does not exist.
            (Explicit is better than implicit.)
        Mention other relevant options.
    notes:
        initialized -> created
    Wrap long lines.

resmo committed:
Add no_gateway_ip option to os_subnet module (#3736)

no_gateway_ip option can provide subnet that is not having a gateway.
fixes #1880

alvaroaleman committed:
Fix can_update test in os_subnet (#3664)

bcoca committed:
deprecated docker (#3710)

* deprecated docker

* updated formatting

Commit:
Fix for issue 3740. Make docker_service more resilient when targeting older API versions.

bcoca committed:
first draft systemd service plugin (#3660)

* first draft systemd service plugin

* now systemd status handles multiline values

dan-petty committed:
fixed $source null check, reference to $Params

$source check causes:
FAILED! => {"changed": false, "failed": true, "msg": "A parameter cannot be found that matches parameter name 'Source'."}

$Params.Remove causes:
FAILED! => {"changed": false, "failed": true, "msg": "Method invocation failed because [System.Management.Automation.PSCustomObject] does not contain a method named 'Remove'."}

gigaroby committed:
Add an example for SRV record in route53 (#3749)

Commit:
Merge pull request #3745 from chouseknecht/fix_3740

Fix for issue 3740. Make docker_service more resilient when targeting…

Commit:
Fix missing key error when updating existing virtual network.

Commit:
Merge pull request #3753 from chouseknecht/fix_azure_virtualnet

Fix missing key error when updating existing virtual network.

bcoca committed:
added state in results for backwards compat

bcoca committed:
fixed typo

bcoca committed:
fixed enabled reporting

powo committed:
Update junos_facts.py (#3759)

abadger committed:
Python2.4's zipfile library cannot handle zip 64bit extensions which are needed for > 64K files (#3754)

Fallback to unzip if zipfile fails and hope that unzip can deal with it
(sites have an easier time upgrading the unzip utility than all of
python).

https://bugs.python.org/issue3997

Fixes #3560

GGabriele committed:
Modified nxos_config (#3522)

* Modified nxos_config

* Resolving conflict

AMDmi3 committed:
Fix crontab argument order for writing (#3750)

Currently, when writing user's crontab, ansible calls

    crontab <file> -u <user>

This is incorrect according to crontab(1) on both FreeBSD and Linux,
which suggest that file argument should be the last.

At least on FreeBSD, this leads to incorrect cron module bahavior which
writes to root's crontab instead of users's

GGabriele committed:
Fixing bug, added missing param (#3737)

davidfischer-ch committed:
Add MariaDB to valid RDS engines (#3337)

* Add MariaDB to valid RDS engines

* Update RDS module documentation

Commit:
Merge pull request #3347 from dougluce/fix-s3-region-error

Fix S3 unavailable region error

kireledan committed:
Added statement about escaping characters in chron tasks (#3770)

danschmidt5189 committed:
Fix "ansbile_facts" typo in docker_container module

dagwieers committed:
win_setup: Add missing properties

We add some basic properties for Windows that are available on other platforms.

dagwieers committed:
Format BIOS date string like on Linux

dagwieers committed:
Added more properties

Commit:
Fix for #3783 - command returns not found or does not exist.

Commit:
Merge pull request #3784 from chouseknecht/fix_3783

Fix for #3783 - command returns not found or does not exist.

Commit:
Fix restart_poicy

Commit:
Update restart_policy doc strings.

Commit:
Merge pull request #3785 from chouseknecht/fix_3779

Fix restart_poicy

Commit:
Fix for issue #3775 - env config comparison

Commit:
Check self.parameters.env exists.

Commit:
Merge pull request #3786 from chouseknecht/fix_3775

Fix for issue #3775 - env config comparison

Commit:
Merge pull request #3776 from danschmidt5189/fix-docker_container-typo

Fix "ansbile_facts" typo in docker_container module

cambid committed:
fix for #3706 (#3778)

Commit:
Fix for 3768 - adding support for env_file

Commit:
Merge pull request #3787 from chouseknecht/fix_3768

Fix for 3768 - adding support for env_file

Commit:
First pass at implementing networks parameter. Also added purge_networks option to remove container from networks not included in networks param.

Commit:
Fixed issues post testing.

Commit:
Fix doc strings.

Commit:
Fixed bug in _get_network_id

Commit:
For new options move vesion_added to 2.1.1

Commit:
Set version added to 2.2 so that tests pass

Shrews committed:
Merge pull request #3732 from GheRivero/devel

Make add ip_floating_ip idempotent

Commit:
Fix network comparison. Fix handling of links. Updated doc strings. Added more examples.

Commit:
Remove debug/register from examples.

Commit:
Merge pull request #3796 from chouseknecht/fix_3766

Fix 3766

anton44eg committed:
Update gc_storage.py (#3802)

isopy commented:
It looks like in my previous test case the KMS ARN that I was referencing wasn't appropriate for an EBS volume. In my case, I was referencing the KMS ARN value for the default aws/redshift KMS key.  I created a KMS key specific for my use and tested the ability to assign this to a newly created EBS volume and it worked perfectly.  I do think that there should better error handling in cases where the KMS ARN referenced isn't appropriate or doesn't exist. Other than that, it's doing what I need it to do.

Commit:
Merge pull request #901 from underyx/assemble-with-prefix-and-suffix

Add prefixing and suffixing fuctionality to assemble module

Commit:
Fix volume to support not binding volumes to host

freym committed:
Fixed error in assemble module when validation failed. (#3814)

Error was: TypeError: fail_json() takes exactly 1 argument (2 given)

Commit:
Fix missing ansible_facts.

Commit:
Fixed binds and volumes. Now recognizes bound vs unbound as well as named volumes.

bcoca committed:
fixed reported state

bcoca committed:
fixed enabled state

Commit:
Merge pull request #3815 from chouseknecht/fix_azure_vm

Fix missing ansible_facts.

bcoca committed:
Revert "Add prefixing and suffixing fuctionality to assemble"

This reverts commit 4c386dba565109d971e88be961abe93d700939b3.
keeping in sync with main repo revert

resmo committed:
apt: fix KeyError for deb=file (#3816)

Fix KeyError: 'prepared' while installing dependencies using deb=<file>.deb
This error shows up when --diff was not passed by and the deb files has dependencies not yet installed.
Closes #3752.

resmo committed:
apt: fix deb=file with --diff does not show diff on upgrades (#3826)

Commit:
Merge pull request #3828 from chouseknecht/winggundamth-devel

Fix volume to support not binding volumes to host

Commit:
Fix upate typo and expected volume compare.

Commit:
Merge pull request #3832 from chouseknecht/devel

Fix upate typo

Commit:
Remove registery option per #3824.

Commit:
Merge pull request #3836 from chouseknecht/devel

Remove registery option per #3824.

Commit:
Fix #3822 stop container

Commit:
Fixing error exception handling for python. Does not need to be compatible with Python2.4 b/c boto is Python 2.6 and above. (#3839)

abadger committed:
Remove .travis.yml from the blacklist

Commit:
Fixing error exception handling for python3. Does not need to be compa… (#3840)

* Fixing error exception handling for python. Does not need to be compatible with Python2.4 b/c boto is Python 2.6 and above.

* Fixing error exception handling for python. Does not need to be compatible with Python2.4 b/c boto is Python 2.6 and above.

* Fixing compile time errors IRT error exception handling for Python 3.5.
This does not need to be compatible with Python2.4 b/c Boto is Python 2.6 and above.

Commit:
Fixing compile time errors irt exception handling for Python 3 (#3841)

* Fixing error exception handling for python. Does not need to be compatible with Python2.4 b/c boto is Python 2.6 and above.

* Fixing error exception handling for python. Does not need to be compatible with Python2.4 b/c boto is Python 2.6 and above.

abadger committed:
Remove modules from py3 blacklist that now compile with PY3

gregdek commented:
Thanks @kalpatel01 for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

Commit:
Fixing Docker compile time errors irt exception handling for Python 3

Commit:
Add support for PostgreSQL 9.5 in rds_param_group (fix for #3846)

Commit:
Fixing compile time error exception handling for python 3. (#3843)

Commit:
Fixing compile time errors irt exception handling for Python 3. This particular diff fixes problems with Exception handling and the use/misues IRT Python 3 of octal numbers (InvalidToken exceptions).

abadger committed:
Digitalocean no longer needs to be on the python3 blacklist

Commit:
Merge pull request #3837 from chouseknecht/devel

Fix #3822 stop container

Commit:
Merge pull request #3844 from Hypermanzer/docker-python-3-compat-bug

Fixing Docker compile time errors irt exception handling for Python 3

Commit:
Remove unused method in ContainerManager

Commit:
Merge pull request #3853 from chouseknecht/devel

Remove unused method in ContainerManager

Commit:
Fix #147 automatically expose published ports

Commit:
Merge pull request #3856 from chouseknecht/devel

Fix #147 automatically expose published ports

Commit:
Fix 3857 name regex

Commit:
Merge pull request #3859 from chouseknecht/devel

Fix 3857 name regex

Commit:
Fixing gce compile time errors irt exception handling for Python 3 (#3845)

Commit:
Fixing open stack compile time errors irt exception handling for Python 3 (#3848)

Commit:
Fixing Rackspace compile time errors irt exception handling for Python 3 (#3849)

abadger committed:
Remove rackspace modules from travis py3 blacklist as they should now compile

nitzmahone committed:
Merge pull request #3345 from mpappas86/bugfix/ec2groupCheckMode

Prevent ec2group from deleting sgs during check_mode runs

ryansb committed:
Merge pull request #2555 from jsternberg/paginate-iam-results

Paginate the results from `boto.iam`

ryansb committed:
Merge pull request #3847 from washingtoneg/postgres9.5-rds-param-group-support

Add support for PostgreSQL 9.5 in rds_param_group (fix for #3846)

nitzmahone committed:
correct gs_storage docs to match reality

Commit:
1) Modified error handling to work with Python 2.4.
2) Fixed octal codes to fall in line with the ansible guide, "Porting Modules to Python 3"
3) updated the requirements.
All changes have been verified against Python 2.4, 2.6, & 3.5.

Commit:
Corrected string of text...

Commit:
Missed umask octal conversion.

Commit:
Fix issue #3866

Commit:
Merge pull request #3867 from chouseknecht/devel

Fix issue #3866

abadger committed:
Be sure to import urllib.parse from somewhere

abadger committed:
Fix the six import

Commit:
corrected the string formatting (format isn't in 2.4 and even in some later versions of python 2.x requires an index). Tested for both compile ***& runtime exceptions*** against 2.4, 2.6, and 3.5.

Commit:
data type fix on format string

Commit:
a) fail_json uses kwargs, b) outside the fork, log instead of returning json on stdout.

bcoca committed:
fixed issue with multiline output from systemd

fixes #3868

senk committed:
Fixed Typo in Examples of systemd module (#3870)

direvus committed:
Fix grammar in docs for fetch 'fail_on_missing' (#3872)

mansunkuo committed:
Add an example to delete Route53 alias record set (#3834)

trondhindenes committed:
Return reboot pending as true/false

nitzmahone committed:
Merge pull request #3879 from trondhindenes/win_fact_reboot_pending

Return reboot pending as true/false

Commit:
removes requirement for config argument to be specified in junos_facts

The config argument is optional and should not be required.  This
removes the required attribute from the config argument.

fixes #3878

privateip committed:
Merge pull request #3886 from privateip/bugfix-3878

removes requirement for config argument to be specified in junos_facts

nitzmahone committed:
Iam trust policy (#3885)

* added support for trust policies

* added version_added to new params in DOC string

* update version_added on new iam args to 2.2

Commit:
move environment variable gathering to end of __init__() (#3890)

* This moves the lines in the code that parse the `env` and `env_file` options for docker to the end of the `__init__()` function.
This is needed because the `_check_capabilites` function needs both a working `self.client` and a proper `self.docker_py_versioninfo`.
`_check_capabilities` is used by `ensure_capabilities` which is, in turn, used by `get_environment`

This means that before this commit, the environment variables could not be loaded because both `self.client` and `self.docker_py_versioninfo` were not set at that time.

This commit fixes that by putting the environment variable parsing after those two.

* This moves the lines in the code that parse the `env` and `env_file` options for docker to the end of the `__init__()` function.
This is needed because the `_check_capabilites` function needs both a working `self.client` and a proper `self.docker_py_versioninfo`.
`_check_capabilities` is used by `ensure_capabilities` which is, in turn, used by `get_environment`

This means that before this commit, the environment variables could not be loaded because both `self.client` and `self.docker_py_versioninfo` were not set at that time.

This commit fixes that by putting the environment variable parsing after those two.

nitzmahone committed:
Merge pull request #3850 from Hypermanzer/utilities_helper_accelerate_p2_p3

Fixing compile time errors irt exception handling for Python 3. This …

dagwieers committed:
Added CPU information comparable to Linux

dagwieers committed:
Ensure we are using an English locale for date formatting.
(as hinted by @nitzmahone)

stefanha committed:
Stream image data in load_image() to avoid out-of-memory

Reading the entire tar file into memory can result in out-of-memory
conditions such as this traceback:

Traceback (most recent call last):
  File "/tmp/ansible_YELTSu/ansible_module_docker_image.py", line 486, in load_image
    self.client.load_image(image_data)
  File "/usr/local/lib/python2.7/dist-packages/docker/api/image.py", line 147, in load_image
    res = self._post(self._url("/images/load"), data=data)
  ...
  File "/usr/lib/python2.7/httplib.py", line 997, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 848, in _send_output
    msg += message_body
MemoryError

Luckily docker-py's load_image(), which calls requests post(), accepts a
file-like object instead of a string.  Pass in the file object to avoid
reading the full file into memory.  This allows larger tar files to load
succesfully.

Signed-off-by: Stefan Hajnoczi <stefanha@gmail.com>

dagwieers committed:
Retain an important note regarding Win32_PhysicalMemory on virtual platforms

dagwieers committed:
Fix ansible_distribution and weekday locale (as hinted by @nitzmahone)

robinro committed:
Fix git shallow update (#3794)

* remove unused variables

* fetch branch name instead of HEAD

fix #3782, which was introduced by f1bacc1d3f9578f26d4ae2f66112cbb2509a7fe8

* disable git depth option for old git versions

fixes #3782
git support for `--depth` did not fully work in old git versions (before 1.8.2)
fall back to full clones/fetches on those versions

jctanner committed:
Use the six import for urlsplit instead of importing directly. (#3902)

Fixes https://github.com/ansible/ansible/issues/16191

abadger committed:
Revert "Fix git shallow update (#3794)"  -- Broke integration tests:
https://app.shippable.com/runs/57599a7897ae890c00c2898d

This reverts commit c26eb1217fae8703a7b9cd8ae708051c58fc9273.

Commit:
added missing colons to documentation (#3913)

dagwieers committed:
Improve the unzip output scraping (#3819)

* Improve the unzip output scraping

Ensure we capture the complete file (also when it includes spaces).
Drop lines that do not conform (in length) to what we expect (e.g. header/footer).

This fixes #3813

* Fix how split() works

dagwieers committed:
Adapt module to use new module._name property (#3918)

This is in line with the change from ansible/ansible#16087

dagwieers committed:
Update GitHub templates to reflect ansible/ansible (#3900)

* Update GitHub templates to reflect ansible/ansible

Update the GitHub templates to what is used for some time on ansible/ansible
For more information, see ansible/ansible#15961

* Small improvement from ansible/ansible

quinot committed:
Clarify behaviour when state=present and no name is given (#3904)

matthewfischer committed:
Update delete notes for Synrhconize docs (#3723)

The example for delete=yes does not specify recursive although it is
required. In addition, the wording for the delete option is confusing
about from where files are really deleted. This should clarify that.

Commit:
Fix #3906 - Enforce required minimum compose version. (#3925)

* Fix #3906 - Require minimum compose version.

* Fix typo in message text.

Commit:
fix unarchive doesn't extract changed tar file #3901

Commit:
Fix #16211 - only stop/kill running containers

Commit:
Merge pull request #3928 from chouseknecht/fix_16211

Fix #3927 - only stop/kill running containers

stefanha committed:
docker_service: make PyYAML requirement explicit

The "Developing Modules" documentation states:

  Include a minimum of dependencies if possible. If there are
  dependencies, document them at the top of the module file, and have
  the module raise JSON error messages when the import fails.

When docker_service runs on a remote host without PyYAML it crashes with
ImportError.

This patch raises a JSON error message when import fails, but only if
the PyYAML module is actually used.  It's only needed when the
"definition" parameter is given.

Signed-off-by: Stefan Hajnoczi <stefanha@gmail.com>

Commit:
Merge pull request #3931 from stefanha/docker_service-PyYAML-requirement

docker_service: make PyYAML requirement explicit

Commit:
Fix #3941

Commit:
Merge pull request #3944 from chouseknecht/devel

avoid removing attached containers by default

Commit:
Fixing compile time errors irt (, e => as e, print(), ocat now 0o not 0) exception handling for Python 3  (#3851)

* Fixing compile time errors irt a) exception handling for Python 3 in util, also: b) problem octal usage (fixed) and c) print json_dump -> print(json_dump(xyz) ... et al

* This code was not Python 2.4 compliant. Octal codes and exception handling is now working with Py 2.4, 2.6, & 3.5.

* Fixing formating (or rather reverting an non 2.4 compatible change). Works in compile & runtime checking.

* a) revert to use print sys.stderr not fail_json; b) fixed var name in exception

* Python 3 compatible print (print >>sys.stderr will generate a TypeError - now uses sys.stderr.write instead).

abadger committed:
Finish python3 syntax compilation fixing for core repo

alikins committed:
Merge pull request #3926 from neowu/devel

fix unarchive doesn't extract changed tar file

alikins committed:
Merge pull request #3701 from gillesgagniard/gce-subnetwork-dev

GCE : Add support for subnet networks

bocharsky-bw committed:
Add mode option to docstring for get_url (#3630)

Add mode option to the list

bcoca committed:
Revert "Add mode option to the list" (#3946)

bcoca committed:
document that get_url also takes file params

has for a long time, but was never documented, mode,group, owner can all be set here

Commit:
Fix #3945 docker_container detach not waiting on container to execute.

Commit:
Remove dns from container create parameters.

Commit:
Merge pull request #3947 from chouseknecht/devel

docker_container detach not waiting on container to execute.

Commit:
Merge pull request #3535 from picpicpic/devel

fixes dns type in docker module

Commit:
Merge pull request #3894 from stefanha/docker_image-load_image-streaming

Stream image data in load_image() to avoid out-of-memory

nitzmahone committed:
rename ansible_winrm_certificate_expires fact to avoid collision with connection vars

bcoca committed:
updated raw docs for environment and executable

it should now reflect how raw works in combination with
become and environment directives

quater committed:
fixed unknown type <type 'list'> error (#3959)

bcoca committed:
restored incorrectly removed docs, jinja2 override

nitzmahone committed:
StrictMode fixes for win_get_url

adambutler committed:
Fixes incorrect key name protocols -> protocol (#3963)

ewigkeit1204 committed:
Do not copy file if running on check mode

abadger committed:
Fix exception handling in async_wrapper

bcoca committed:
exception ans sys.exit fixes

bcoca committed:
added missing import

readded raise

nitzmahone committed:
Merge pull request #3965 from ewigkeit1204/fix-copy-module

Do not copy file if running on check mode

bcoca committed:
reverted need for module utils imports

XooR committed:
Fix wait_for on python < 2.6 (#3790) (#3973)

jmunhoz committed:
Add proper version_added for rgw option in s3.py

Signed-off-by: Javier M. Mellid <jmunhoz@igalia.com>

jmunhoz committed:
Add s3_url requirement in doc when rgw support is enabled in s3.py

Signed-off-by: Javier M. Mellid <jmunhoz@igalia.com>

gregdek commented:
@kalpatel01 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

mariojmdavid committed:
add project to os_router (#3869)

* add project to os_router

* version_added fix

* correct place of version_added

* remove extraneous else

* * add shade version dependency check
* option project only on router creation

* * fix shade minimal version
* remove pop of option project

* * fix square brackets
* delete router by router_id

Commit:
Fixes #1715 Allow authorized_module accept multi. (#3573)

This will allow the authorized_module to accept options that can be
passed multiple times into ssh options. For instance permitopen.

Commit:
Fix ulimit comparison

Commit:
Fix index

Commit:
Fix compose_version error. Check has compose before checking version.

Commit:
Merge pull request #3995 from chouseknecht/fix_3978

Fix compose_version error. Check has compose before checking version.

Commit:
Merge pull request #3994 from chouseknecht/fix_ulimit

Fix ulimit comparison

mgedmin committed:
git: fix change testing when version is HEAD (#3025)

When `version` is not specified, it defaults to "HEAD".  "HEAD" is not a
remote tag, and it's not listed in the output of get_branches(), so we'd
keep repo_updated at the default value (None) and then return early with
changed=True in --check mode, even when before == after.

Fixes #3024.

abadger committed:
Add comment

jctanner committed:
Add a note about using the script module with pseudo-tty's lack of stderr channel (#4000)

Addresses #3610

ryansb committed:
Merge pull request #3091 from jmunhoz/s3-ceph

Add Ceph RGW S3 compatibility

jctanner committed:
Add local and url support to yum state=latest (#4003)

Fixes #3807

mattymo committed:
Add retries to apt_key with keyserver, fixes #3986 (#3987)

Public SKS gpg servers frequently are unavailable, but a retry
can mitigate frequent failures.

G-Goldstein committed:
Change documented options for os_networks_facts (#3970)

* Change documented options for os_networks_facts

os_network_facts currently lists 'network' as an available option, taking the Name or ID. In Ansible 2.0.2 to 2.2.0, this is not valid. Options 'name' and 'id' should be used instead.

* Update os_networks_facts.py

* Update os_networks_facts.py

Set version_added to the only accepted value

* Update os_networks_facts.py

Removed inappropriate 'ID' parameter

kireledan committed:
apt remove now recognizes the force parameter (#3756)

jctanner committed:
Cleanup unused variable in yum.py (#4009)

bcoca committed:
removed extra space

as pointed out by @lloydbenson

nitzmahone committed:
Merge pull request #3747 from mixmatch/devel

fixed $source null check, reference to $Params

concaf committed:
Fix broken link for issue template (#3680)

Commit:
Fixed delete statement for juniper_config.py #4007 #3984

privateip committed:
Merge pull request #4031 from stianvi/devel

Fixed delete statement for juniper_config.py #4007 #3984

dagwieers committed:
Comment out $win32_cs.SystemFamily, cfr. @nitzmahone

Since it fails on everything pre-10/2016

bcoca committed:
updated docs in pause to reflect 0 value behaviour

nitzmahone committed:
Merge pull request #3777 from dagwieers/win_setup-updates

win_setup: Add missing properties

abadger committed:
A few more sanity checks for detecting unzip output that's not a file entry (#3982)

* A few more sanity checks for detecting unzip output that's not a file entry

Also note that there's a rounding error somewhere in the mtime
comparison code.

* Fix reference to sub-array

bcoca committed:
made password no_log in htpasswd

Commit:
Update and pin to azure-2.0.0rc5

nitzmahone committed:
fixes to windows setup

coerce multi-socket win32_processor to a single value
WindowsIdentity.Label doesn't exist on .NET 4.0

talonx committed:
Fix for #16518 - added missing regions

ryansb committed:
Remove double-assignment of EC2 parameters (#4081)

The `source_dest_check` and `termination_protection` variables are being
assigned twice in ec2.py, likely due to an incorrect merge somewhere
along the line.

nitzmahone committed:
Merge pull request #4080 from talonx/devel

Fix for #16518 - added missing regions

tongpu committed:
also match on tabs in ini_file (fixes #106) (#4067)

also match on tabs in ini_file

mlosapio committed:
Showing some of the sysctl settings in proper yaml format per Ansible documentation (#4073)

alikins committed:
Fix default perm for apt_repo files. (#4072)

Change the file mode arg to 'raw' ala file args

Following the file_common_args model, change the
type of the 'mode' arg here to type='raw' with no
default arg value.

The default mode for file creation is the module
constant DEFAULT_SOURCES_PER, and is used if no
mode os specified.

A default mode of 0644 (and not specified as int or str)
would get converted to an octal 420, resulting in the
sources file being created with mode '0420' instead of '0644'

Fixes #16370

alikins committed:
Add 'force_register' to redhat_subscriptions (#3976)

Fixes #2448

alikins committed:
Update docs and internal naming s/RHN/RHSM (#3977)

puiterwijk committed:
nova_compute: indicate argument type (#4070)

Without this, ansible 2.1 will convert some arguments that are
meant to be dict or list type to their str representation.

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

bcoca committed:
allow pip to take a list of names (#4056)

also simplified argspec by removing defaults

Commit:
Merge pull request #4097 from chouseknecht/azure-rc5

Update and pin to azure-2.0.0rc5

gregdek commented:
@kalpatel01 Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

bcoca committed:
make daemon reload first thing always run

otherwise service detection can fail before we run it which might
be required to actually detect the service.

problame committed:
service module: use sysrc on FreeBSD (#4042)

* service module: use sysrc on FreeBSD

sysrc(8) is the designated userland program to edit rc files on FreeBSD.

It first appeared in FreeBSD 9.2, hence is available on all supported
versions of FreeBSD.

Side effect: fixes #2664

* Incorporate changes suggested by bcoca.

- Use `get_bin_path` to find sysrc binary.
- Only use sysrc when available (support for legacy versions of FreeBSD)

Commit:
Merge branch 'devel' of https://github.com/kalpatel01/ansible-modules-core into devel

kalpatel01 commented:
I'll reopen, merged caused a mess
